lib/babe/babe.go

```diff
@@ -336,77 +301,73 @@ func (b *Service) initiate() {
 		return
 	}
 
-	err := b.invokeBlockAuthoring()
+	// we should consider better error handling for this - we should
+	// retry to run the engine at some point (maybe the next epoch) if
+	// there's an error.
+	if err := b.runEngine(); err != nil {
+		logger.Criticalf("failed to run block production engine: %s", err)
```

Comment:
shouldn't the whole node shut down in this case?

Reply:
probably, so should i panic here?

Reply:
I'd rather not panic.  Can we not signal the node to shutdown? This would require us to refactor the `Service` interface to return an `err` through an `errChan`, and the node could then shutdown knowing that `babe.Service` had failed.

Reply:
This 💯